mitodl/micromasters
# refactor course enrollments

#### What are the relevant tickets?

part of #3057 

#### What's this PR do?

This refactors the course enrollments front-end API code to use the redux-rest abstraction we wrote recently.

#### How should this be manually tested?

Make sure you can add a course enrollment, and also that the failure case is handled well.
